sql/wsrep_mysqld.cc

```diff
@@ -1181,6 +1181,51 @@ void wsrep_keys_free(wsrep_key_arr_t* key_arr)
     key_arr->keys_len= 0;
 }
 
+void
+wsrep_append_fk_parent_table(THD* thd, TABLE_LIST* tables, wsrep::key_array* keys)
+{
+  if (!WSREP(thd) || !WSREP_CLIENT(thd)) return;
+    TABLE_LIST *table;
+
+    thd->mdl_context.release_transactional_locks();
+    uint counter;
+    MDL_savepoint mdl_savepoint= thd->mdl_context.mdl_savepoint();
+
+    bool open_error=
+      open_tables(thd, &tables, &counter, MYSQL_OPEN_FORCE_SHARED_HIGH_PRIO_MDL);
+    if (unlikely(open_error && (thd->killed || thd->is_error())))
+    {
+      WSREP_WARN("unable to open table for FK checks in OPTIMIZE processing");
```

Comment:
If you the call to `open_table()` fails a warning is printed. However, there is no `return` from here, and the process goes on to iterate over the tables... Shouldn't this be handled somehow?